prov/gni/src/gnix_ep.c

```diff
@@ -749,10 +745,15 @@ static void __ep_destruct(void *obj)
 		_gnix_ref_put(av);
 
 	/*
-	 * clean up any vc hash table or vector
+	 * clean up any vc hash table or vector,
+	 * remove entry from addr_to_ep ht.
 	 */
 
 	if (ep->type == FI_EP_RDM) {
+
+		key_ptr = (gnix_ht_key_t *)&ep->my_name.gnix_addr;
+		ret =  _gnix_ht_remove(ep->cm_nic->addr_to_ep_ht,
+				       *key_ptr);
 		if (ep->vc_ht != NULL) {
 			_gnix_ht_destroy(ep->vc_ht);
```

Comment:
In the next line:

     free(ep->vc_ht);

Not sure what the freeing strategy should be, but probably should warn.